File: Backend/src/main/java/com/businessexchange/business/entity/Business.java
package com.businessexchange.business.entity;

import com.businessexchange.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "businesses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Business {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "seller_id", nullable = false)
    private User seller;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private BusinessCategory category;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, length = 200)
    private String location;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(name = "asking_price", nullable = false)
    private BigDecimal askingPrice;

    @Column(name = "business_age_years")
    private Integer businessAgeYears;

    @Column(name = "number_of_employees")
    private Integer numberOfEmployees;

    @Column(name = "reason_for_selling", columnDefinition = "TEXT")
    private String reasonForSelling;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private BusinessStatus status = BusinessStatus.PENDING_REVIEW;

    @ManyToOne
    @JoinColumn(name = "approved_by")
    private User approvedBy;

    @Column(name = "approved_at")
    private LocalDateTime approvedAt;

    @Column(name = "rejection_reason", columnDefinition = "TEXT")
    private String rejectionReason;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();

        if (status == null) {
            status = BusinessStatus.PENDING_REVIEW;
        }
    }

    @PreUpdate
    public void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: Frontend/src/types/business.ts
export interface BusinessFile {
  id: number;
  fileType: "IMAGE" | "DOCUMENT" | "FINANCIAL_REPORT";
  originalName: string;
  url: string;
  uploadedAt: string;
}

export interface Business {
  id: number;
  title: string;
  category: string;
  sellerName: string;
  description: string;
  location: string;
  askingPrice: number;
  status: string;
  rejectionReason?: string;
  isFavorited?: boolean;
}

export interface BusinessApiResponse {
  success: boolean;
  message: string;
  data: Business[];
}

export interface SingleBusinessApiResponse {
  success: boolean;
  message: string;
  data: Business;
}

export interface BusinessCreateRequest {
  sellerId: number;
  category: string;
  title: string;
  description: string;
  location: string;
  address?: string;
  askingPrice: number;
  businessAgeYears?: number;
  numberOfEmployees?: number;
  reasonForSelling?: string;
}